### PR TITLE
Fix incentive reward autocompound

### PR DESCRIPTION
**Pull request by Buckram123**
This PR aims to fix incentive reward autocompound for cases, where incentive is not one of the 2 pool assets.
Beyond that it includes
- Version bump
- Schema update
- Removal of token0/token1(`denom`) in favor of always using asset0/asset1(`AssetEntry`)

### CODE REVIEW

File: contracts/carrot-app/src/handlers/execute.rs [hunk 1]
```diff
@@ -181,8 +181,13 @@ fn autocompound(deps: DepsMut, env: Env, info: MessageInfo, app: App) -> AppResu
 
     // If there are external incentives, claim them.
     if !position.claimable_incentives.is_empty() {
+        let asset0_denom = position.asset0.unwrap().denom;
+        let asset1_denom = position.asset1.unwrap().denom;
+
         for coin in try_proto_to_cosmwasm_coins(position.claimable_incentives)? {
-            rewards.add(coin)?;
+            if coin.denom == asset0_denom || coin.denom == asset1_denom {
```
Comment (Kayanski) [inline contracts/carrot-app/src/handlers/execute.rs:188]: Wait, we're not counting other reward types as rewards ? This feels weird

File: contracts/carrot-app/tests/autocompound.rs [hunk 2]
```diff
@@ -10,15 +11,48 @@ use carrot_app::msg::{
...
 
 #[test]
 fn check_autocompound() -> anyhow::Result<()> {
-    let (_, carrot_app) = setup_test_tube(false)?;
+    let (pool_id, carrot_app) = setup_test_tube(false)?;
 
     let chain = carrot_app.get_chain().clone();
 
+    // Add incentive
+    {
```
Comment (Kayanski) [inline contracts/carrot-app/tests/autocompound.rs:28]: Nice !

**Review by Kayanski — COMMENTED** [review]
Don't really understand the logic behind not counting non asset0 and asset2 rewards towards rewards

File: contracts/carrot-app/src/handlers/execute.rs (hunk 1 above)
Comment (Buckram123) [reply contracts/carrot-app/src/handlers/execute.rs:188]: We don't autocompound them, because it's not possible to add uosmo to usdc/usdt pool

Comment (Kayanski) [reply contracts/carrot-app/src/handlers/execute.rs:188]: Ah ok we don't swap those tokens yet. Got it. Adding that in v2

File: bot/src/bot.rs [hunk 3]
```diff
@@ -220,10 +220,12 @@ fn autocompound_instance(daemon: &Daemon, instance: (&str, &Addr)) -> anyhow::Re
     app.set_address(address);
     use carrot_app::AppQueryMsgFns;
     let resp: CompoundStatusResponse = app.compound_status()?;
+    let pool_rewards: AvailableRewardsResponse = app.available_rewards()?;
```
Comment (CyberHoward) [inline bot/src/bot.rs:223]: We should combine both queries into one so we only have to do one RPC request.

File: contracts/carrot-app/src/state.rs [hunk 4]
```diff
@@ -62,8 +62,6 @@ impl AutocompoundRewardsConfig {
 #[cw_serde]
 pub struct PoolConfig {
     pub pool_id: u64,
-    pub token0: String,
```
Comment (CyberHoward) [inline contracts/carrot-app/src/state.rs:65]: **Requires migrate fn**

**Review by CyberHoward — COMMENTED** [review]
LGTM apart from a missing migrate fn to account for the state changes!

File: bot/src/bot.rs (hunk 3 above)
Comment (Buckram123) [reply bot/src/bot.rs:223]: > We should combine both queries into one so we only have to do one RPC request.

Sure

File: contracts/carrot-app/src/state.rs (hunk 4 above)
Comment (Buckram123) [reply contracts/carrot-app/src/state.rs:65]: Added that, didn't think we planned to merge any contract

File: bot/src/bot.rs (hunk 3 above)
Comment (Buckram123) [reply bot/src/bot.rs]: Added that!

**Review by CyberHoward — APPROVED** [review]

**Pull request merged**